src/tests/fs_tests.rs

```diff
@@ -4036,4 +4037,136 @@ pub mod fs_tests {
         assert_eq!(cage.exit_syscall(EXIT_SUCCESS), EXIT_SUCCESS);
         lindrustfinalize();
     }
+
+    pub fn ut_lind_fs_stat_syscall_tests() {
+        //acquiring a lock on TESTMUTEX prevents other tests from running concurrently,
+        // and also performs clean env setup
+        let _thelock = setup::lock_and_init();
+
+        let cage = interface::cagetable_getref(1);
+
+        let mut statdata = StatData::default();
+
+        // test out whether an error is output for a non existent file path
+        // (ENOENT[-2])
+        assert_eq!(
+            cage.stat_syscall("non_existent_file_path", &mut statdata),
+            syscall_error(Errno::ENOENT, "stat", "test_failure")
```

Comment:
I think only checking with -(Errno::ENOENT as i32) suffices for checking errors - we don't need to call the syscall_error function explicitly

Reply:
ENOENT is a positive value, whereas the errors returned are negative. I thought instead of typecasting and negating the result, it would be clearer to understand this way